packages/apps/graph/src/graph/Subscription/newBlocks.ts

```diff
@@ -28,13 +29,14 @@ async function* iteratorFn(
   chainIds: number[] = Array.from(new Array(dotenv.CHAIN_COUNT)).map(
     (__, i) => i,
   ),
+  context: IContext,
 ): AsyncGenerator<Block[], void, unknown> {
   let lastBlock = (await getLastBlocks(chainIds))[0];
 
   yield [lastBlock];
   log('yielding initial block with id %s', lastBlock.id);
 
-  while (true) {
+  while (!context.req.socket.destroyed) {
```

Comment:
Noticed that the loop continues if no block has been fetched and the subscription ended